Babylon/commands/abba/run.py

```diff
@@ -0,0 +1,194 @@
+    service_state = state["services"]
+    detailed_data = []
+    if 'duration' not in df.columns:
+        df['duration'] = ""
+    for i, entry in enumerate(rows):
+        service_state["api"]["organization_id"] = entry.get('organizationId')
+        service_state["api"]["workspace_id"] = entry.get("workspaceId")
+        service_state["api"]["scenariorun_id"] = entry.get("scenariorunId")
+        response = get_scenariorun_status(service_state, azure_token)
+        start_time = pd.to_datetime(response['startTime'])
+        end_time = pd.to_datetime(response['endTime'])
+        df.loc[i, 'duration'] = (end_time - start_time).total_seconds()
+        if response.get('phase') == "Succeeded":
+            detailed_data.append(summarize(response, i))
+        else:
+            logger.info(f"Scenariorun {entry.get('scenariorunId')} has phase {response.get('phase')}")
+    generate_report(df, detailed_data)
+    return CommandResponse.success()
+
+
+def get_scenariorun_status(service_state, azure_token):
+    # get scenariorun status
+    service = ScenarioRunService(state=service_state, azure_token=azure_token)
+    response = service.status()
+    if response is None:
+        logger.error(f"Failed to get status of {service_state['api']['scenariorun_id']}")
+    return response.json()
+
+
+def summarize(data: dict, i: int):
+    df = pd.DataFrame.from_records(data['nodes'])[['containerName', 'startTime', 'endTime']]
+    df['startTime'] = pd.to_datetime(df['startTime'])
+    df['endTime'] = pd.to_datetime(df['endTime'])
+    df['duration'] = (df['endTime'] - df['startTime']).dt.total_seconds()
+    logger.info(f"Report {i} generated")
+    return (df)
+
+
+def generate_report(summary_df, detailed):
+    """Generate summary report from simulation run data."""
+    overall_plot = go.Figure(data=[go.Bar(x=summary_df['duration'], y=summary_df['scenariorunId'], orientation='h')])
+    overall_plot.update_layout(title='Scenariorun Durations')
+    top = pyo.plot(overall_plot, include_plotlyjs=False, output_type='div')
+    fig_list = []
+    for i, detailed_data in enumerate(detailed):
+        fig = px.timeline(detailed_data,
+                          title=f"Detailed report for {'i'}",
+                          x_start="startTime",
+                          x_end="endTime",
+                          y="containerName")
+        fig.update_traces(text=detailed_data['duration'], textposition='outside')
+        fig.update_layout(title="Execution time by step (seconds)", xaxis_title="Time", yaxis_title="Step")
+        fig_list.append(pyo.plot(fig, include_plotlyjs=False, output_type='div'))
+    # Generate static HTML page (should be viewed online)
+    html_content = f"""
+  <!DOCTYPE html>
+  <html>
+  <head>
+    <title>Simulation Report</title>
+  </head>
+  <body>
+    <script src="https://cdn.plot.ly/plotly-latest.min.js"></script>
+    <h1>Total execution time</h1>
+    {top}
+    <h1>Execution time by step</h1>
```

Comment:
This is just front-end dev speaking :rofl: we cannot have two h1 tags in one html tree :slightly_smiling_face: